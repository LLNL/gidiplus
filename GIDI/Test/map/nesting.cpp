/*
# <<BEGIN-copyright>>
# Copyright 2019, Lawrence Livermore National Security, LLC.
# See the top-level COPYRIGHT file for details.
# 
# SPDX-License-Identifier: MIT
# <<END-copyright>>
*/

#include <stdlib.h>
#include <iostream>

#include "GIDI_testUtilities.hpp"

void walk( PoPI::Database &pops, char const *fileName );
bool mapWalkCallBack( GIDI::Map::ProtareBase const *a_protareEntry, std::string const &a_library, void *a_data, int a_level );
void printList( char const *prefix, std::vector<std::string> &list );
/*
=========================================================
*/
int main( int argc, char **argv ) {

    PoPI::Database pops( "../pops.xml" );

    printCodeArguments( __FILE__, argc, argv );

    if( argc > 1 ) throw "No arguments are allowed";

    walk( pops, "neutrons1.map" );
    walk( pops, "neutrons2.map" );
    walk( pops, "all.map" );
    walk( pops, "level3.map" );
    walk( pops, "level4.map" );

    exit( EXIT_SUCCESS );
}
/*
=========================================================
*/
void walk( PoPI::Database &pops, char const *fileName ) {

    GIDI::Map::Map *map;

    try {
        map = new GIDI::Map::Map( fileName, pops ); }
    catch ( char const *str ) {
        std::cout << str << std::endl;
        exit( EXIT_FAILURE );
    }

    std::cout << std::endl;
    std::cout << "library = " << map->library( ) << std::endl;
    std::cout << std::endl;

    map->walk( mapWalkCallBack, map );

    delete map;
}
/*
=========================================================
*/
bool mapWalkCallBack( GIDI::Map::ProtareBase const *a_protareEntry, std::string const &a_library, void *a_data, int a_level ) {

    std::string path( stripDirectoryBase( a_protareEntry->path( ) ) );

    for( int i1 = 0; i1 < a_level + 1; ++i1 ) std::cout << "    ";
    std::cout << "path = " << path << ":: evaluation = " << a_protareEntry->evaluation( ) << ":: libraries are: ";
    std::vector<std::string> libraries;
    a_protareEntry->libraries( libraries );
    for( std::vector<std::string>::iterator iter = libraries.begin( ); iter != libraries.end( ); ++iter ) std::cout << " " << *iter;
    std::cout << std::endl;

    return( true );
}
/*
=========================================================
*/
void printList( char const *prefix, std::vector<std::string> &list ) {

    std::string sep( "" );

    std::cout << prefix << " <";
    for( std::vector<std::string>::const_iterator iter = list.begin( ); iter != list.end( ); ++iter ) {
        std::cout << sep << "'" << *iter << "'";
        sep = ", ";
    }
    std::cout << ">" << std::endl;
}
