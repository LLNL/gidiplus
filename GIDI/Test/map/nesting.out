
library = neutrons1

    path = /.../GIDI/Test/map/neutrons/n-001_H_001.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-008_O_016.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-018_Ar_036.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-022_Ti_049.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-028_Ni_061.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-033_As_074.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-035_Br_079.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-039_Y_089.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-042_Mo_098.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-046_Pd_105.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-049_In_113.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-051_Sb_124.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-053_I_135.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-056_Ba_140.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-060_Nd_145.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-062_Sm_154.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-066_Dy_156.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-071_Lu_175.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-073_Ta_181.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-080_Hg_204.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-090_Th_230.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-092_U_234.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-092_U_236.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-092_U_238.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-092_U_239.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-094_Pu_242.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-095_Am_244.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-096_Cm_247.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1
    path = /.../GIDI/Test/map/neutrons/n-100_Fm_255.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1

library = neutrons2

    path = /.../GIDI/Test/map/neutrons/n-001_H_001.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-008_O_017.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-018_Ar_036.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-028_Ni_061.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-035_Br_079.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-042_Mo_098.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-049_In_113.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-053_I_135.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-060_Nd_145.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-066_Dy_156.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-073_Ta_181.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-090_Th_230.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-092_U_236.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-092_U_239.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2
    path = /.../GIDI/Test/map/neutrons/n-095_Am_244.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2

library = all

        path = /.../GIDI/Test/map/neutrons/n-001_H_001.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-008_O_016.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-018_Ar_036.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-022_Ti_049.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-028_Ni_061.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-033_As_074.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-035_Br_079.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-039_Y_089.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-042_Mo_098.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-046_Pd_105.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-049_In_113.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-051_Sb_124.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-053_I_135.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-056_Ba_140.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-060_Nd_145.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-062_Sm_154.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-066_Dy_156.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-071_Lu_175.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-073_Ta_181.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-080_Hg_204.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-090_Th_230.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-092_U_234.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-092_U_236.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-092_U_238.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-092_U_239.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-094_Pu_242.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-095_Am_244.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-096_Cm_247.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-100_Fm_255.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all
        path = /.../GIDI/Test/map/neutrons/n-001_H_001.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-008_O_017.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-018_Ar_036.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-028_Ni_061.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-035_Br_079.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-042_Mo_098.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-049_In_113.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-053_I_135.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-060_Nd_145.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-066_Dy_156.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-073_Ta_181.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-090_Th_230.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-092_U_236.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-092_U_239.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all
        path = /.../GIDI/Test/map/neutrons/n-095_Am_244.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all

library = level3

    path = /.../GIDI/Test/map/neutrons/n-008_O_016.xml:: evaluation = ENDF/BVII.1:: libraries are:  level3
            path = /.../GIDI/Test/map/neutrons/n-001_H_001.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-008_O_016.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-018_Ar_036.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-022_Ti_049.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-028_Ni_061.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-033_As_074.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-035_Br_079.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-039_Y_089.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-042_Mo_098.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-046_Pd_105.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-049_In_113.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-051_Sb_124.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-053_I_135.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-056_Ba_140.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-060_Nd_145.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-062_Sm_154.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-066_Dy_156.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-071_Lu_175.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-073_Ta_181.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-080_Hg_204.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-090_Th_230.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-092_U_234.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-092_U_236.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-092_U_238.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-092_U_239.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-094_Pu_242.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-095_Am_244.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-096_Cm_247.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-100_Fm_255.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3
            path = /.../GIDI/Test/map/neutrons/n-001_H_001.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-008_O_017.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-018_Ar_036.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-028_Ni_061.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-035_Br_079.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-042_Mo_098.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-049_In_113.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-053_I_135.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-060_Nd_145.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-066_Dy_156.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-073_Ta_181.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-090_Th_230.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-092_U_236.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-092_U_239.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
            path = /.../GIDI/Test/map/neutrons/n-095_Am_244.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3
    path = /.../GIDI/Test/map/neutrons/n-008_O_017.xml:: evaluation = ENDF/BVII.1:: libraries are:  level3

library = level4

    path = /.../GIDI/Test/map/neutrons/n-008_O_017.xml:: evaluation = ENDF/BVII.1:: libraries are:  level4
        path = /.../GIDI/Test/map/neutrons/n-008_O_016.xml:: evaluation = ENDF/BVII.1:: libraries are:  level3 level4
                path = /.../GIDI/Test/map/neutrons/n-001_H_001.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-008_O_016.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-018_Ar_036.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-022_Ti_049.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-028_Ni_061.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-033_As_074.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-035_Br_079.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-039_Y_089.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-042_Mo_098.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-046_Pd_105.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-049_In_113.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-051_Sb_124.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-053_I_135.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-056_Ba_140.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-060_Nd_145.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-062_Sm_154.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-066_Dy_156.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-071_Lu_175.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-073_Ta_181.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-080_Hg_204.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-090_Th_230.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-092_U_234.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-092_U_236.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-092_U_238.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-092_U_239.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-094_Pu_242.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-095_Am_244.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-096_Cm_247.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-100_Fm_255.xml:: evaluation = ENDF/BVII.1:: libraries are:  neutrons1 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-001_H_001.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-008_O_017.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-018_Ar_036.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-028_Ni_061.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-035_Br_079.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-042_Mo_098.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-049_In_113.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-053_I_135.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-060_Nd_145.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-066_Dy_156.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-073_Ta_181.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-090_Th_230.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-092_U_236.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-092_U_239.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
                path = /.../GIDI/Test/map/neutrons/n-095_Am_244.xml:: evaluation = ENDF/BVIII.1:: libraries are:  neutrons2 all level3 level4
        path = /.../GIDI/Test/map/neutrons/n-008_O_017.xml:: evaluation = ENDF/BVII.1:: libraries are:  level3 level4
